server/embed/serve.go

```diff
@@ -301,7 +302,22 @@ func (sctx *serveCtx) registerGateway(dial func(ctx context.Context) (*grpc.Clie
 	if err != nil {
 		return nil, err
 	}
-	gwmux := gw.NewServeMux()
+
+	gwmux := gw.NewServeMux(
+		gw.WithMarshalerOption(gw.MIMEWildcard,
+			&gw.HTTPBodyMarshaler{
+				Marshaler: &gw.JSONPb{
+					MarshalOptions: protojson.MarshalOptions{
+						UseProtoNames:   true,
+						EmitUnpopulated: false,
+					},
+					UnmarshalOptions: protojson.UnmarshalOptions{
+						DiscardUnknown: true,
+					},
+				},
+			},
+		),
+	)
```

Comment:
Does the [defaultMarshaler](https://github.com/grpc-ecosystem/grpc-gateway/blob/e894bd71ea4092e9bd8f9a541f832619e1c9b4cb/runtime/marshaler_registry.go#L20) work?

Reply:
Unfortunately, it doesn't.

If we use [defaultMarshaler](https://github.com/grpc-ecosystem/grpc-gateway/blob/v2.17.0/runtime/marshaler_registry.go#L20) , for the TestV3CurlTxn case, we will receive the response like

```
{
  "header": {
    "clusterId": "3801629817145504659",
    "memberId": "47555275972384877",
    "revision": "2",
    "raftTerm": "2"
  },
  "succeeded": true,
  "responses": [
    {
      "responsePut": {
        "header": {
          "clusterId": "0",
          "memberId": "0",
          "revision": "2",
          "raftTerm": "0"
        },
        "prevKv": null
      }
    }
  ]
}
```

In v1.16.0, the default jsonpb will use json tag and ignore the empty value. https://github.com/grpc-ecosystem/grpc-gateway/blob/v1.16.0/runtime/marshaler_registry.go#L19